target/linux/ramips/dts/mt7621_asus_rp-ac56.dts

```diff
@@ -0,0 +1,231 @@
+		};
+
+		wlan5g_red {
+			label = "rp-ac56:red:wlan5g";
+			gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g_blue {
+			label = "rp-ac56:blue:wlan5g";
+			gpios = <&gpio1 11 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g_green {
+			label = "rp-ac56:green:wlan5g";
+			gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_power_white: power_white {
+			label = "rp-ac56:white:power";
+			gpios = <&gpio1 16 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_power_orange: power_orange {
+			label = "rp-ac56:orange:power";
+			gpios = <&gpio1 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_power_blue: power_blue {
+			label = "rp-ac56:blue:power";
+			gpios = <&gpio1 14 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,name = "Audio-I2S";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,bitclock-master = <&dailink0_master>;
+		simple-audio-card,frame-master = <&dailink0_master>;
+		simple-audio-card,routing =
+			"Headphone Jack", "HP_L",
+			"Headphone Jack", "HP_R";
+		simple-audio-card,mclk-fs = <256>;
+
+		simple-audio-card,cpu {
+			sound-dai = <&i2s>;
+		};
+
+		dailink0_master: simple-audio-card,codec {
+			sound-dai = <&codec>;
+		};
+	};
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "asus,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfa0000>;
+			};
+
+			radio: partition@ff0000 {
+				label = "radio";
+				reg = <0xff0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "uart2", "rgmii2";
+			ralink,function = "gpio";
+		};
+	};
+	i2s_pins: i2s {
+		i2s {
+			ralink,group = "uart3";
+			ralink,function = "i2s";
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "pci14c3,7603";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "pci14c3,7662";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Have you checked whether there are valid addresses in 0x28/0x2e/0xe000/0xe006?

Do the MAC addresses match the assignment in stock firmware?

Further reading: https://openwrt.org/docs/guide-developer/mac.address

Reply:
The original FW is reusing MAC from wlan0 for eth0 so I am doing the same (wlan0 MAC is @0x0004 and wlan1 MAC is @0x8004)

Reply:
Still, the same MAC address could be in one of the mentioned locations, which are the recommended ones. Please check.

Reply:
Sorry, did not mentioned in my original comment: obviously I have  checked it: "factory" partition has only two MACs in it, like I've mentioned before, @0x0004 offset and @0x8004 offset. 
root@mcrpac1:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
**mtd2: 00010000 00010000 "factory"**

root@mcrpac1:~# hexdump -C -s 0x28 /dev/mtd2
00000028  fd 9b ff ff ff ff ff ff  ff ff ff ff 22 34 0c 20  |............"4. |
root@mcrpac1:~# hexdump -C -s 0x2e /dev/mtd2
0000002e  ff ff ff ff ff ff 22 34  0c 20 ff ff 00 01 00 00  |......"4. ......|
root@mcrpac1:~# hexdump -C -s 0xe000 /dev/mtd2
0000e000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
root@mcrpac1:~# hexdump -C -s 0xe006 /dev/mtd2
0000e006  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|

Reply:
Okay. Please preserve that information in the commit message.

Reply:
Done

Reply:
Thanks, two typos:

"This board has only two MAC addresses programed in "factory" parition:"
-> "programmed", "partition"

I'd add a fourth point mentioning:
- No valid addresses found in 0x28, 0x2e, 0xe000 and 0xe006